src/System.Private.CoreLib/shared/System/ValueTuple.cs

```diff
@@ -2093,7 +2097,7 @@ int IStructuralEquatable.GetHashCode(IEqualityComparer comparer)
 
         private int GetHashCodeCore(IEqualityComparer comparer)
         {
-            // We want to have a limited hash in this case.  We'll use the last 8 elements of the tuple
+            // We want to have a limited hash in this case. We'll use the first 7 elements of the tuple
```

Comment:
The same refactoring to compute restHashCode once and store it in a local can be applied here as well.

Reply:
Ok. I didn't do the cast change out of fear that I'd actually increase allocations rather than reduce them if the JIT optimization didn't kick in. But I can do the local thing for symmetry.